tests/file_open_test.c

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include "fcntl.h"
+#include <aws/common/file_open.h>
+
+#include <aws/testing/aws_test_harness.h>
+
+static int s_aws_fopen_test_helper(char *file_path, char *content) {
+    char read_result[100];
+    AWS_ZERO_ARRAY(read_result);
+    FILE *file = aws_fopen(file_path, "w+");
+    ASSERT_NOT_NULL(file);
+    fprintf(file, "%s", content);
+    fclose(file);
+    FILE *readfile = aws_fopen(file_path, "r");
+    ASSERT_NOT_NULL(readfile);
+    size_t read_len = fread(read_result, sizeof(char), strlen(content), readfile);
+    ASSERT_UINT_EQUALS(strlen(content), read_len);
+    fclose(readfile);
+    ASSERT_SUCCESS(strcmp(content, read_result));
+
+#ifdef _WIN32
+    wchar_t w_file_path[1000];
+    /* plus one for the EOS */
+    size_t file_path_len = strlen(file_path) + 1;
+    MultiByteToWideChar(CP_UTF8, 0, file_path, -1, w_file_path, (int)file_path_len);
```

Comment:
I don't like this... It's still use the same function we want to check it's correct or not... But, I don't know what else can we do to check the result... An alternative I can think of is to use `[WideCharToMultiByte](WideCharToMultiByte )`, feels like it's still basically the same...

Reply:
you could just commit a file with a utf-8 name to our repo and have the test open THAT

same way we have files committed to `aws-c-io/tests/resources/*`and the aws-c-io tests use those

Reply:
also, you don't need to worry about cleaning up the test files.
we have tons of tests that create local domain sockets and just leave the files on disk, in the test bin directory

Reply:
on second thought... do clean up the files so running the results of one test run don't interfere the NEXT time you run the tests. Those domain sockets are OK because the OS makes sure they're uniquely named.

or randomize the filename to prevent collisions, use [tmpname()](https://en.cppreference.com/w/c/io/tmpnam) or something

Reply:
sounds good, I'll add another test to read an existing file, and keep those tests for testing writing